### PR TITLE
CSV Generation: support path detection for arrayFieldGroups

### LINKED ISSUE #2700: arrayFieldGroups require [0] in path for correct form display in OKD
## Bug Report

**What did you do?**
```golang
type MemcachedSpec struct {
	Volumes []MemcachedVolume `json:"volumes"`
}

type MemcachedVolume struct {
	// Persistent Volume Claim
	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="Persistent Volume Claim"
	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.x-descriptors="urn:alm:descriptor:com.tectonic.ui:arrayFieldGroup:volumes,urn:alm:descriptor:io.kubernetes:PersistentVolumeClaim"
	// +optional
	Claim string `json:"claim,omitempty"`
}
```

**What did you expect to see?**
```yaml
      specDescriptors:
      - description: Persistent Volume Claim
        displayName: Persistent Volume Claim
        path: volumes[0].claim
        x-descriptors:
        - urn:alm:descriptor:com.tectonic.ui:arrayFieldGroup:volumes
        - urn:alm:descriptor:io.kubernetes:PersistentVolumeClaim
```

**What did you see instead? Under which circumstances?**
```yaml
      specDescriptors:
      - description: Persistent Volume Claim
        displayName: Persistent Volume Claim
        path: volumes.claim
        x-descriptors:
        - urn:alm:descriptor:com.tectonic.ui:arrayFieldGroup:volumes
        - urn:alm:descriptor:io.kubernetes:PersistentVolumeClaim
```

**Environment**
* operator-sdk version:

```
operator-sdk version: "v0.16.0-dirty", commit: "55f1446c5f472e7d8e308dcdf36d0d7fc44fc4fd", go version: "go1.13.8 linux/amd64"
```

* go version:

```
go version go1.14.1 linux/amd64
```

* Kubernetes version information:

```
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.0-4-g38212b5", GitCommit:"2576e48", GitTreeState:"clean", BuildDate:"2020-03-05T14:38:53Z", GoVersion:"go1.13.4", Compiler:"gc"
, Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"17+", GitVersion:"v1.17.1", GitCommit:"e18216f", GitTreeState:"clean", BuildDate:"2020-03-20T07:55:19Z", GoVersion:"go1.13.8", Compiler:"gc", Platform
:"linux/amd64"}
```

```
Client Version: 4.4.0-0.okd-2020-03-21-112849
Server Version: 4.4.0-0.okd-2020-03-21-112849
Kubernetes Version: v1.17.1
```

* Kubernetes cluster kind: `OKD 4.4.0-0.okd-2020-03-21-112849`

* Are you writing your operator in ansible, helm, or go? `go`

### PR DESCRIPTION
**Pull request by joelanford**
**Description of the change:**
Add support for path detection of arrayFieldGroup spec descriptors during CSV generation

**Motivation for the change:**
Closes #2700

### CODE REVIEW

File: test/test-framework/pkg/apis/cache/v1alpha1/dummy_types.go [hunk 1]
```diff
@@ -102,6 +107,16 @@ type Engine struct {
...
+	// Type should be in spec with path equal to wheels[0].type
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="Wheel Type"
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.x-descriptors="urn:alm:descriptor:com.tectonic.ui:arrayFieldGroup:wheels,urn:alm:descriptor:com.tectonic.ui:text"
+	Type string `json:"type"`
+}
+
 // +k8s:deepcopy-gen=false
 // +k8s:openapi-gen=false
 type InlinedComponent struct {
```
Comment (camilamacedo86) [inline test/test-framework/pkg/apis/cache/v1alpha1/dummy_types.go]: Missing update the file gen with. 
Could you please run `./hack/generate/gen-test-framework.sh` to update the mock data as well?

**Comment by joelanford** [comment]
@camilamacedo86 

> Could you please run ./hack/generate/gen-test-framework.sh to update the mock data as well?
:+1:

> Also, see that it will not gen automatically the CSV, so, could we gen that to check its result? Could we add in the script as well to gen it automatically?
I ran this and noticed that the CSVs in the test framework are pretty out-of-date. To check the result, I did add unit tests.

@estroz Do you know if the CSVs in the test-framework are actually used for anything? If not, should they be removed?

**Comment by estroz** [comment]
@camilamacedo86 I believe they're being used for scorecard tests. @jmccormick2001 confirm?

**Comment by jmccormick2001** [comment]
deploy/olm-catalog/memcached-operator/0.0.3/memcached-operator.v0.0.3.clusterserviceversion.yaml

that one is being used by the scorecard tests currently.

**Comment by joelanford** [comment]
Maybe let's do a follow-up PR to update that CSV and remove the others?

**Pull request merged**